app/src/main/java/com/quran/labs/androidquran/ui/PagerActivity.java

```diff
@@ -493,8 +496,29 @@ public void onPageSelected(int position) {
     downloadReceiver.setListener(this);
   }
 
-  private void saveRecentPage(int page) {
-    recentPagePresenter.onPageChanged(page);
+  public Observable<Integer> getViewPagerObservable() {
```

Comment:
some "thinking out loud" thoughts:

i am thinking something like this should be used to also clean up other logic (our massive `onCreate` method in `PagerActivity` for example. thinking about splitting some of that logic into a `PagePresenter` of sorts, and that can also `getViewPagerObservable` and subscribe to it so as to handle changes there separately from here.

at that point, we can then decide to cache this Observable in a variable initialized in onCreate and calling `share` on it so we can have multiple observers to it (with only one `OnPageChangeListener`).

another thought:
should we split out presenters and some sort of "view presenters" - i.e. what's the right place for this method to live? it felt strange for this to be `getViewPager` instead and the work be done in the presenter.